pkg/build/controller/controller_test.go

```diff
@@ -390,14 +399,11 @@ func TestHandleBuild(t *testing.T) {
 		if tc.inStatus != buildapi.BuildPhaseError && build.Status.Phase == buildapi.BuildPhaseError && len(build.Status.Message) == 0 {
 			t.Errorf("(%d) errored build should set message: %#v", i, build)
 		}
-		if len(tc.outputSpec) != 0 {
-			build := ctrl.BuildStrategy.(*okStrategy).build
```

Comment:
Are you sure they point to the same object in memory?
You can print the pointers with `fmt.Printf("%[1]p %[1]T, %[2]p %[2]T\n", ref1, ref2)`, where the refs should be of type `*Build`.